common/endpoints/endpoints.go

```diff
@@ -64,3 +77,97 @@ func MakeStatsReceiver(scope StatScope) stats.StatsReceiver {
 		15*time.Second)
 	return s.Scope(string(scope))
 }
+
+const StdoutName = "stdout"
+const StderrName = "stderr"
+
+func NewResourceHandler(uri string, ajaxInterval time.Duration) *ResourceHandler {
+	hostname, _ := os.Hostname()
+	return &ResourceHandler{resources: make(map[string]map[string]string), hostname: hostname, uri: uri, ajaxInterval: ajaxInterval}
+}
+
+type ResourceHandler struct {
+	// Defines: map[Namespace]map[ResourceName]ResourcePath
+	resources    map[string]map[string]string
```

Comment:
Some comments on the resource handler would be good here... Is it just to serve the JS?